Splitting the projects into deployable packages

tdeborge opened the pull request:
added some logic so that packages can now be deployed in a separate fashion. For this, I added the common libraries as a dp as they are reused by the camel and simulator packages. So now you can choose to install all by the master package or individual in the different sub-projects.

tdeborge committed:
Worked on the project and added dp package builds in all the sub-projects. This allows to deploy the packages separately (like the simulator/sensor tag).
I thought it would make more sense like this ...

ccustine commented:
There are two things I would like to see in order to accept this PR.  If you could update so that it doesn't include the Eclipse project files that would help a lot, as well as not update whitespace in the existing files.  I added entries in .gitignore for IDEA, and I will do the same for Eclipse files to make this easier in the future. As for the whitespace, I noticed a couple of license headers were really malformed due to whitespace adjustment, and it affected a large number of existing source files as well. Generally speaking, I already have the source files formatted according to the Eclipse Kura project formatting guidelines.

If you can make these updates I will see about merging.

tdeborge commented:
Sure thing ... learning to become a contributor :) ... quick pointer to the formatting guidelines?

Is there a style file to do the auto formatting?

Verstuurd vanaf mijn iPhone

> Op 1 nov. 2016 om 09:48 heeft Chris Custine notifications@github.com het volgende geschreven:
> 
> There are two things I would like to see in order to accept this PR. If you could update so that it doesn't include the Eclipse project files that would help a lot, as well as not update whitespace in the existing files. I added entries in .gitignore for IDEA, and I will do the same for Eclipse files to make this easier in the future. As for the whitespace, I noticed a couple of license headers were really malformed due to whitespace adjustment, and it affected a large number of existing source files as well. Generally speaking, I already have the source files formatted according to the Eclipse Kura project formatting guidelines.
> 
> If you can make these updates I will see about merging.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

ccustine commented:
@tdeborge There is an Eclipse formatter definition in the Kura project here: https://github.com/eclipse/kura/blob/develop/kura/setups/formatting/KuraFormatter.xml

I imported that into IDEA so some of the setting may not match exactly, but this project should be close. Thanks for looking at this!

tdeborge commented:
Will do asap ... came down with a respiratory infection combined with fever ...

Thanks for the pointer

Verstuurd vanaf mijn iPhone

> Op 1 nov. 2016 om 20:10 heeft Chris Custine notifications@github.com het volgende geschreven:
> 
> @tdeborge There is an Eclipse formatter definition in the Kura project here: https://github.com/eclipse/kura/blob/develop/kura/setups/formatting/KuraFormatter.xml
> 
> I imported that into IDEA so some of the setting may not match exactly, but this project should be close. Thanks for looking at this!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

tdeborge committed:
adding gitignore information for eclipse and mac

tdeborge committed:
Deleting all Eclipse related files

tdeborge committed:
Deleting all Eclipse related files

tdeborge committed:
Adding bin to the ignore list (as this is a building directory)

tdeborge committed:
Merge branch 'master' into Cleanup

* master:
  adding gitignore information for eclipse and mac

# Conflicts:
#	.gitignore
#	.project

tdeborge committed:
Merge branch 'master' into Cleanup

* master:
  adding gitignore information for eclipse and mac

# Conflicts:
#	.gitignore
#	.project
Deleting all Eclipse related files

tdeborge committed:
Reformatted the java classes according to the Kura formatting template

tdeborge commented:
I removed all Eclipse related information
I updated the .gitignore to include the eclipse specific files
I reformatted the java files according to the Kura Formatter
